src/components/directory-content-table.tsx

```diff
@@ -64,7 +68,49 @@ export const DirectoryContentTable = ({
         },
         [getRowStyle]
     );
+    const dispatch = useDispatch();
+    const columnOrder = useSelector((state: AppState) => state.reorderedColumns); // Adjust according to your state structure
 
+    // Initialize columnDefs based on columnOrder from Redux
+    useEffect(() => {
+        // Extract column order from colDef if columnOrder from Redux is not available
+        const extractColumnOrder = (columnDefs: ColDef[]): string[] => {
+            return columnDefs.filter((col) => col.field).map((col) => col.field as string);
+        };
+
+        if (!columnOrder || columnOrder.length === 0) {
+            const initialColumnOrder = extractColumnOrder(colDef);
+            dispatch(setReorderedColumns(initialColumnOrder));
+        } else {
+            const orderedColumnDefs = reorderColumns(colDef, columnOrder);
+            setColumnDefs(orderedColumnDefs);
+        }
+    }, [columnOrder, colDef, dispatch]);
+
+    const reorderColumns = (colDef: ColDef[], newFieldOrder: string[] | undefined): ColDef[] => {
+        const fieldIndexMap = new Map(newFieldOrder?.map((field, index) => [field, index]));
+        return colDef
+            .filter((col) => fieldIndexMap.has(col.field || ''))
+            .sort((a, b) => {
+                const indexA = fieldIndexMap.get(a.field || '') ?? -1;
+                const indexB = fieldIndexMap.get(b.field || '') ?? -1;
+                return indexA - indexB;
+            });
+    };
```

Comment:
this function 'renderColumns' can be defined outside the component so it's will be created only once for the entire application. because the function does not need any state or props